### PR TITLE
SASS-9593: Integrated Interest Task List endpoint

### PR DESCRIPTION
**Pull request by ansahdavid**
### Description
Please include a brief summary of the change you've made and the issue that it fixes.

Add a link to the relevant story in Jira
Main Ticket - SASS-8104

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [ ]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [ ]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests? (where applicable)
- [ ]  Have you addressed warnings where appropriate?
- [ ]  Have you rebased against the current version of main?
- [ ]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [ ]  Have you checked the PR Builder passes?

### CODE REVIEW

File: app/connectors/InterestTaskListDataConnector.scala [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class InterestTaskListDataConnector @Inject()(val http: HttpClient, val config: AppConfig)
+                                             (implicit ec: ExecutionContext) extends Connector {
+
+  def get(taxYear: Int,nino:String)(implicit hc: HeaderCarrier): Future[TaskListSectionResponseModel] = {
+
+    val taskListDataUrl: String = config.interestBaseUrl + s"/income-tax-interest/income-tax/$taxYear/tasks/$nino"
```
Comment (ghousedev) [inline app/connectors/InterestTaskListDataConnector.scala:31]: The url should be: `s"/income-tax-interest/$taxYear/tasks/$nino"`.

Comment (ansahdavid) [reply app/connectors/InterestTaskListDataConnector.scala]: Thanks for point this out, made the change in the next commit.

File: test/services/TaskListDataServiceSpec.scala [hunk 2]
```diff
@@ -62,11 +62,18 @@ class TaskListDataServiceSpec extends AnyWordSpec with Matchers with ScalaFuture
...
-    TaskListSection(CharitableDonationsTitle, Some(Seq(
+    TaskListSection(
+      CharitableDonationsTitle, Some(Seq(
       TaskListSectionItem(DonationsUsingGiftAid, TaskStatus.Completed, Some("CYAPage")),
       TaskListSectionItem(GiftsOfLandOrProperty, TaskStatus.Completed, Some("CYAPage")),
       TaskListSectionItem(GiftsOfShares, TaskStatus.Completed, Some("CYAPage")),
       TaskListSectionItem(GiftsToOverseas, TaskStatus.Completed, Some("CYAPage"))
+    ))),
+    TaskListSection(
+      InterestTitle, Some(Seq(
```
Comment (basotia) [inline test/services/TaskListDataServiceSpec.scala:73]: Hi David, Could you please use below testing scenario as an example. As both the tailoring service and Interest service returning the same response. will always match, we need to make sure what happens when there aer different responses 
`    TaskListSection(
      InsuranceGainsTitle, Some(Seq(
        TaskListSectionItem(LifeInsurance, TaskStatus.InProgress, Some("LifeInsurancePage")),
        TaskListSectionItem(LifeAnnuity, TaskStatus.Completed, Some("CYAPage")),
        TaskListSectionItem(CapitalRedemption, TaskStatus.NotStarted, Some("CapitalRedemptionPage")),
        TaskListSectionItem(VoidedISA, TaskStatus.NotStarted, Some("VoidedISAPage"))
      )))`

Comment (ansahdavid) [reply test/services/TaskListDataServiceSpec.scala:73]: Hi Manish, I think this maybe a case of me simply removing the `InterestTitle` TaskListSection from here as it has its own response and the `InsuranceGainsTitle` TaskListSection is already present in this response.

**Review by basotia — APPROVED** [review]

**Pull request merged**